### PR TITLE
[hue] make sure that hue in HSBType is never >=360

### PR DESCRIPTION
**Pull request by kaikreuzer**
Fixes https://github.com/eclipse/smarthome/issues/6788

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/LightStateConverterTest.java [hunk 1]
```diff
@@ -101,6 +101,17 @@ public void hsbBrightnessAlwaysGreaterThanZero() {
         }
     }
 
+    @Test
+    public void hsbHueAlwaysGreaterThanZeroAndLessThan360() {
+        final State lightState = new State();
+        lightState.colormode = ColorMode.CT.toString();
```
Comment (cweitkamp) [inline extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/LightStateConverterTest.java]: We maybe want to use `ColorMode.HS.toString()` in this case too - which is default and you do not need to set it.

Comment (kaikreuzer) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/LightStateConverterTest.java]: changed, thx.

File: extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java [hunk 2]
```diff
@@ -189,7 +187,7 @@ public void assertCommandForColorChannelRed() {
 
     @Test
     public void assertCommandForColorChannelBlue() {
-        String expectedReply = "{\"bri\" : 254, \"sat\" : 254, \"hue\" : 43680}";
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: @cweitkamp Do you happen to know whether 43680 was fixed value for blue (i.e. given in the hue docs)?
I've adapted it to what the new code uses and would assume that this is the correct way.

Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: I do not know if you can access it without account:

https://developers.meethue.com/develop/hue-api/lights-api/#get-attr-and-state

| hue | uint16 | Hue of the light. This is a wrapping value  between 0 and 65535. Note, that hue/sat values are hardware dependent  which means that programming two devices with the same value does not  garantuee that they will be the same color. Programming 0 and 65535  would mean that the light will resemble the color red, 21845 for green  and 43690 for blue. |
| -- | -- | -- |

Comment (kaikreuzer) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: Which means that the new value is definitely better than the old one 😎 .

Comment (kaikreuzer) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: And with [this commit](https://github.com/eclipse/smarthome/pull/6790/commits/4c95ecf6a0e954e8137188187e473650d0db947c) it now even hits the exact value!

Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: Awesome. Does it match green too? Red already works fine.

Comment (kaikreuzer) [reply extensions/binding/org.eclipse.smarthome.binding.hue.test/src/test/java/org/eclipse/smarthome/binding/hue/internal/handler/HueLightHandlerTest.java]: [Works as well.](https://github.com/eclipse/smarthome/pull/6790/commits/8d776aea7b686f80c9220fe53d74f0ea444bb182)

**Pull request merged**